faux-mgs/src/main.rs

```diff
@@ -1135,8 +1147,15 @@ async fn run_command(
                 ]))
             }
         }
-        Command::ComponentActiveSlot { component, set, persist } => {
-            if let Some(slot) = set {
+        Command::ComponentActiveSlot { component, set, persist, transient } => {
+            if transient && component != SpComponent::ROT {
+                let cmd = Args::command();
+                clap::Error::raw( clap::error::ErrorKind::ArgumentConflict,
+                    format!(
+                        "The --transient (-t) flag is only allowed for the 'rot' component, not for {}\n", component
+                        )
+                    ).with_cmd(&cmd).exit();
```

Comment:
I don't think doing the clap parsing gets us anything over just doing `bail!` which matches what we do elsehwere in the file. I also really wish there was a way to specify this requirement in clap directly.

Reply:
You are right, it just gets us a capital E on "Error: ..." instead of lower case "error".